File: ruby/wesley_safadao.rb
class WesleySafadao
  def initialize(day, month, year)
    @day = day.to_i
    @month = month.to_i
    @year = year.to_i
  end

  def safadeza
    return (1..@month).reduce(:+) + (@year / 100.to_f) + (50 - @day) 
  end

  def angel
    return 100 - safadeza()
  end

  def say_safadeza
    puts "Você é #{safadeza.round(2)}% safado e #{angel.round(2)}% anjo"
  end
end

File: ocaml/WesleySafadao.ml
let getPercentualSafadeza (day, month, year) = 
    (1+month)*month/2 + (year/100) * (50 - day)

let getPercentualAnjo (day, month, year) = 
    100 - getPercentualSafadeza (day, month, year)
